rules/gsuite_activityevent_rules/google_workspace_new_login_ip.py

```diff
@@ -0,0 +1,111 @@
+import json
+import ast
+from datetime import datetime, timedelta
+
+from panther_base_helpers import deep_get
+from panther_oss_helpers import get_string_set, put_string_set
+
+ALERT_CONTEXT_DICTIONARY = {}
+
+# Store the record in Dynamo for each user, for this amount of days
+# If the user does not log in within this period, the record will expire, and the next login will alert
+DYNAMO_CACHE_DAYS = 30
+
+
+def rule(event):
+    # see: https://developers.google.com/admin-sdk/reports/v1/appendix/activity/login#login
+    is_login_event = event.get("type") == 'login' and event.get("name") == 'login_success'
+    if not is_login_event:
+        return False
+
+    ip_address = get_ip_address(event)
+    user_identifier = get_user_identifier(event)
+
+    if not ip_address or not user_identifier:
```

Comment:
user_identifier will never be null because there is a default

Reply:
It might still technically be falsy, i don't know how panther's default condition works (is it none or falsy?)
I will just remove the default though, as getting alerts for "unknown user" aren't helpful

Reply:
in this case, user_identifier would = "<Unknown user>" which would evaluate as "true". For strings only an empty string evaluates as false

Reply:
I meant that it might be possible for event.get to treat an empty string return value as not falsy (thus not trigger its default), but it doesn't seem likely. It is best to just avoid the default setting regardless